eng/pipelines/tasks/CSharp.yml

```diff
@@ -0,0 +1,22 @@
+steps:
+- task: NuGetToolInstaller@0
+  inputs:
+    versionSpec: '5.8.0'
+
+- task: NuGetCommand@2
```

Comment:
BuildSolution.yml already calls NuGet Restore.
https://github.com/microsoft/MIEngine/blob/220ecfd07fa672ee966083ff70c140863cad1970/eng/pipelines/steps/BuildSolution.yml#L9-L14

If you move `CSharp.yml` after `BuildSolution.yml` in `Build.template.yml` you will get the same results instead of duplicating task runs.